### ISSUE TITLE
float with decimal not sorted properly

### ISSUE THREAD

**Issue by jd-0001** [typesense/typesense#114]
## Description

I have datetime field which has float type and can have values in decimals. When I try to sort it result doesn't consider decimals(milliseconds)

## Steps to reproduce

![Screenshot from 2020-06-12 16-23-29](https://user-images.githubusercontent.com/47495003/84495631-1a6d5c00-acc9-11ea-8d69-882c3d664281.png)
![Screenshot from 2020-06-12 16-23-43](https://user-images.githubusercontent.com/47495003/84495637-1c371f80-acc9-11ea-893a-1dc52217cb07.png)

## Expected Behavior

I think it should also sort the data using decimals

## Actual Behavior

It looks like it ignores decimal and sort the data

## Metadata

**Typsense Version**: 0.13.0

**OS**: Linux Mint

**Comment by jd-0001** [comment]
I just converted that float field to int64 and used timestamp in miliseconds to handle sorting properly. 

Sorting works fine for now but if I query on that field server get crashed:
```python
# Works!
search_parameters = {
    'q': '*',
    'query_by': 'item_name',
    'per_page': 21,
    'sort_by': 'approved_date:desc',
    # 'filter_by': 'approved_date:<1591091288061',
    'include_fields': 'approved_date,item_name'
}
```

But If I introduce filter:
```python
# Server Crahses
search_parameters = {
    'q': '*',
    'query_by': 'item_name',
    'per_page': 21,
    'sort_by': 'approved_date:desc',
    'filter_by': 'approved_date:<1591091288061',
    'include_fields': 'approved_date,item_name'
}
```

**Comment by kishorenc** [comment]
IEEE-754 floating point numbers use 24 bits for mantissa and 8-bit for exponent. So the largest float value that be stored precisely is `16777216`. Anything more than that (timestamps in this case) will only be "approximately" represented. So, using an int64 datatype for the timestamp is the correct approach. 

To find out what's going wrong with the filter, can you please give me:

a) a few sample documents, which after insertion triggers the crash during filter
b) the full crash stack trace you see
c) schema of the collection (you can use generic field names like X, Y etc. if you don't want to reveal exact names)

**Comment by jd-0001** [comment]
Hi @kishorenc I uploaded your requirements in drive: [view](https://drive.google.com/file/d/1svGv12eg3nEvDUvhUvcDU38_pNqgnnmI/view?usp=sharing)

**Comment by kishorenc** [comment]
Thank you, will take a look and update this thread.

**Comment by kishorenc** [comment]
@jd-0001 This is fixed in https://github.com/typesense/typesense/commit/a744f6e566cabfa18eef6777bac852dacf5989bc and the fix is available in this Docker image: `typesense/typesense:a744f6e`

If you are using an actual Linux executable for local dev and would prefer that, I will also be happy to create a custom build for you -- let me know.

**Comment by jd-0001** [comment]
Yes, I am using linux executable. I will test it and will let you know.

Thanks :pray:

**Comment by jd-0001** [comment]
Hi, @kishorenc When I will get this update? or can I get executable you were referring to so I can test your solution?

**Comment by kishorenc** [comment]
Sorry, I thought you were able to test via the Docker image. Please use this binary for testing the change:

https://dl.typesense.org/releases/nightly/typesense-server-a744f6e-linux-amd64.tar.gz

**Comment by kishorenc** [comment]
Fix for filtering on int64 values is available in `v0.14.0` just released.
